tests/infrastructure/tekton/test_tekton_pipeline_disk_uploader.py

```diff
@@ -0,0 +1,168 @@
+    {"name": PUSH_TIMEOUT, "type": "string"},
+    {"name": SECRET_NAME, "type": "string"},
+]
+
+pipeline_tasks = [
+    {
+        "name": DISK_UPLOADER_TASK,
+        "params": [
+            {"name": EXPORT_SOURCE_KIND, "value": f"$(params.{EXPORT_SOURCE_KIND})"},
+            {"name": EXPORT_SOURCE_NAME, "value": f"$(params.{EXPORT_SOURCE_NAME})"},
+            {"name": VOLUME_NAME, "value": f"$(params.{VOLUME_NAME})"},
+            {"name": IMAGE_DESTINATION, "value": f"$(params.{IMAGE_DESTINATION})"},
+            {"name": PUSH_TIMEOUT, "value": f"$(params.{PUSH_TIMEOUT})"},
+            {"name": SECRET_NAME, "value": f"$(params.{SECRET_NAME})"},
+        ],
+        "taskRef": {"kind": "Task", "name": DISK_UPLOADER_TASK},
+    }
+]
+
+
+@pytest.fixture()
+def quay_disk_uploader_secret(custom_pipeline_namespace):
+    with Secret(
+        name="quay-disk-uploader-secret",
+        namespace=custom_pipeline_namespace.name,
+        accesskeyid=base64_encode_str(os.environ["QUAY_ACCESS_KEY_TEKTON_TASKS"]),
+        secretkey=base64_encode_str(os.environ["QUAY_SECRET_KEY_TEKTON_TASKS"]),
+    ) as quay_disk_uploader_secret:
+        yield quay_disk_uploader_secret
+
+
+@pytest.fixture()
+def created_fedora_dv(admin_client, golden_images_namespace, custom_pipeline_namespace):
+    with create_dv(
+        dv_name="fedora-dv-disk-uploader",
+        namespace=custom_pipeline_namespace.name,
+        source=PVC,
+        source_pvc=DataSource(
+            name=OS_FLAVOR_FEDORA, namespace=golden_images_namespace.name
+        ).instance.spec.source.pvc.name,
+        source_namespace=golden_images_namespace.name,
+        size="35Gi",
+        client=admin_client,
+        storage_class=py_config["default_storage_class"],
+    ) as dv:
+        dv.wait_for_dv_success()
+        yield dv
+
+
+@pytest.fixture()
+def vm_for_disk_uploader(custom_pipeline_namespace, admin_client, created_fedora_dv):
+    with VirtualMachineForTests(
+        name="fedora-vm-diskuploader",
+        namespace=custom_pipeline_namespace.name,
+        client=admin_client,
+        data_volume=created_fedora_dv,
+        vm_instance_type=VirtualMachineClusterInstancetype(name="u1.small"),
+        vm_preference=VirtualMachineClusterPreference(name=OS_FLAVOR_FEDORA),
+        run_strategy=VirtualMachine.RunStrategy.ALWAYS,
+    ) as vm:
+        wait_for_running_vm(vm=vm, wait_for_interfaces=False, check_ssh_connectivity=False)
+        yield vm
+
+
+@pytest.fixture()
+def stopped_vm_for_disk_uploader(vm_for_disk_uploader):
+    vm_for_disk_uploader.stop(wait=True)
+    yield
+
+
+@pytest.fixture()
+def pipeline_disk_uploader(
+    admin_client,
+    custom_pipeline_namespace,
+):
+    with Pipeline(
+        name="pipeline-disk-uploader",
+        namespace=custom_pipeline_namespace.name,
+        client=admin_client,
+        tasks=pipeline_tasks,
+        params=pipeline_params,
+    ) as pipeline:
+        if pipeline.exists:
```

Comment:
this is redundant. you'll anyway get an error if pipeline is not created

Reply:
@vsibirsk  do you mean removing if pipeline.exists and keep only below?
`) as pipeline:
     yield pipeline`

Reply:
correct

Reply:
+1 to what Vasiliy said.